worker/src/main.rs

```diff
@@ -188,31 +197,88 @@ fn worker(node_url: &str, w_ip: &str, w_port: &str, mu_ra_port: &str) {
 
 	// ------------------------------------------------------------------------
 	// start the substrate-api-client to communicate with the node
-	let mut api = Api::new(format!("ws://{}", node_url));
-	api.init();
+	let alice = primitives::sr25519::Pair::from_string("//Alice", None).unwrap();
+	println!("   Alice's account = {}", alice.public().to_ss58check());
+	let alicekey = AccountKey::Sr(alice.clone());
+	// alice is validator
+	let mut api = Api::new(format!("ws://{}", node_url))
+	   	.set_signer(alicekey.clone());
+
+	info!("encoding Alice public 	= {:?}", alice.public().0.encode());
+	info!("encoding Alice AccountId = {:?}", AccountId::from(alice.public()).encode());
+
+	let result_str = api.get_storage("Balances", "FreeBalance", Some(AccountId::from(alice.public()).encode())).unwrap();
+    let funds = hexstr_to_u256(result_str).unwrap();
+	println!("    Alice's free balance = {:?}", funds);
+    let result_str = api.get_storage("System", "AccountNonce", Some(AccountId::from(alice.public()).encode())).unwrap();
+    let result = hexstr_to_u256(result_str).unwrap();
+    println!("    Alice's Account Nonce is {}", result.low_u32());
+
 
 	// ------------------------------------------------------------------------
 	// get required fields for the extrinsic
-	let genesis_hash = api.genesis_hash.unwrap().as_bytes().to_vec();
+	let genesis_hash = api.genesis_hash.as_bytes().to_vec();
 
 	// get the public signing key of the TEE
-	let mut key = [0; 32];
-	let ecc_key = fs::read(ECC_PUB_KEY).expect("Unable to open ECC public key file");
-	key.copy_from_slice(&ecc_key[..]);
-	info!("[+] Got ECC public key of TEE = {:?}", key);
+	println!("*** Ask the signing key from the TEE");
+	let tee_pubkey_size = 32;
+	let mut tee_pubkey = [0u8; 32];
+
+	let mut retval = sgx_status_t::SGX_SUCCESS;
+	let result = unsafe {
+		get_ecc_signing_pubkey(enclave.geteid(),
+							   &mut retval,
+							   tee_pubkey.as_mut_ptr(),
+							   tee_pubkey_size
+		)
+	};
+	match result {
+		sgx_status_t::SGX_SUCCESS => {},
+		_ => {
+			error!("[-] ECALL Enclave Failed {}!", result.as_str());
+			return;
+		}
+	}	
+	// Attention: this HAS to be sr25519, although its a ed25519 key!
+	let tee_public = sr25519::Public::from_raw(tee_pubkey);
+	info!("[+] Got ed25519 account of TEE = {}", tee_public.to_ss58check());
+	//info!("[+] Got ed25519 public raw of  TEE = {:?}", tee_pubkey);
+	let tee_accountid = AccountId::from(tee_public);
+
+	// check the enclave's account balance
+	let result_str = api.get_storage("Balances", "FreeBalance", Some(tee_accountid.encode())).unwrap();
+    let funds = hexstr_to_u256(result_str).unwrap();
+	info!("TEE's free balance = {:?}", funds);
+
+	if funds < U256::from(10) {
+		println!("[+] bootstrap funding Enclave form Alice's funds");
+		let xt = transfer(api.clone(), GenericAddress::from(tee_pubkey), 1000000);
+		let mut _xthex = hex::encode(xt.encode());
+		_xthex.insert_str(0, "0x");
+		let xt_hash = api.send_extrinsic(_xthex).unwrap();
```

Comment:
we can just use the xt.hex_encode() method from the extrinsic

Reply:
cool